gui/src/installer/step/descriptor.rs

```diff
@@ -176,6 +200,14 @@ impl Step for DefineDescriptor {
     }
 }
 
+fn check_key_network(key: &DescriptorPublicKey, network: Network) -> bool {
+    match key {
+        DescriptorPublicKey::XPub(key) => key.xkey.network == network,
+        DescriptorPublicKey::MultiXPub(key) => key.xkey.network == network,
+        _ => true,
+    }
```

Comment:
It's not that simple, unfortunately. For regtest or signet the expected xpub network is testnet. See how it's done in minisafed:
https://github.com/revault/minisafe/blob/5595e91e5700e17a692890b691affa7ffd039fdc/src/config.rs#L199-L216